src/wallet/wallet_rpc_server.cpp

```diff
@@ -105,24 +113,56 @@ namespace tools
     return epee::http_server_impl_base<wallet_rpc_server, connection_context>::run(1, true);
   }
   //------------------------------------------------------------------------------------------------------------------------------
-  bool wallet_rpc_server::init(const boost::program_options::variables_map& vm)
+  void wallet_rpc_server::stop()
   {
-    auto rpc_config = cryptonote::rpc_args::process(vm);
+    if (m_wallet)
+    {
+	  m_wallet->store();
+	  delete m_wallet;
+	  m_wallet = NULL;
+    }
+  }
+  //------------------------------------------------------------------------------------------------------------------------------
+  bool wallet_rpc_server::init(const boost::program_options::variables_map *vm)
```

Comment:
Why change from reference to pointer ? I don't see anywhere obviously passing NULL.